src/formatting.rs

```diff
@@ -2,19 +2,33 @@
 //!
 //! It has functions to format slice of bytes and some help functions to separate concerns while doing the job.
 use crate::config::Config;
-use log::{info, trace};
+use log::{error, info, trace};
 use nu_parser::{flatten_block, parse, FlatShape};
 use nu_protocol::{
     ast::Block,
-    engine::{self, StateWorkingSet},
+    engine::{EngineState, StateWorkingSet},
     Span,
 };
 
+struct DeclId;
+
+#[allow(non_upper_case_globals)]
+impl DeclId {
+    const If: usize = 22;
+    const Let: usize = 30;
+    const Def: usize = 5;
+    const ExportDefEnv: usize = 6;
```

Comment:
these values come from Nushell itself, right?

Reply:
Yes, they come from nushell, but I didn't found them on the repo (I am convinced that they are there, but I didn't find them).
I made that list in base of trial and error, writing down the keywords that gave me problems and reading the log to see what happens

Reply:
mm yeah, true, i don't see them in the code base of Nushell :thinking:

Reply:
we can use these for now, and if we see a more robust way to specify them later... :wink:

Reply:
I don't think you can depend on those IDs being constant. effectively they are generated by when a particular builtin is pushed into the working_set. So if someone would rearrange the `create_default_context()` (or whatever it is called now for `nu-cmd-lang`) this would break.

With the vision for `new-nu-parser` that those are clear keywords and maybe even don't have an underlying command things could become easier.

Maybe as a bridging solution we could first introduce a `FlatShape::KeywordCall` or similar on the Nu-side that has a guaranteed mapping in the flattening.

Reply:
That sounds great!
I actually should write the formatter based on the new parser, shouldn't I? The new parser is on course and will eventually come. Of course because it's not finished there may be some tokens or kewords I could not format yet, but a least with very simple cases I can build a formatter and adapt in the way, what do you think?